src/oneD/StFlow.cpp

```diff
@@ -344,190 +354,259 @@ void StFlow::updateProperties(size_t jg, double* x, size_t jmin, size_t jmax)
+                            double rdt, size_t jmin, size_t jmax)
+{
+    // The left boundary has the same form for all cases.
+    if (jmin == 0) { // left boundary
+        rsd[index(c_offset_U,jmin)] = -(rho_u(x,jmin + 1) - rho_u(x,jmin))/m_dz[jmin]
+                                      -(density(jmin + 1)*V(x,jmin + 1)
+                                      + density(jmin)*V(x,jmin));
+        diag[index(c_offset_U,jmin)] = 0; // Algebraic constraint
+    }
 
-            // set the radiative heat loss vector
-            m_qdotRadiation[j] = radiative_heat_loss;
+    if (jmax == m_points - 1 ) { // right boundary
+        if (m_usesLambda == true) { // axisymmetric flow
+            rsd[index(c_offset_U, jmax)] = rho_u(x, jmax);
+        } else { // right boundary (same for unstrained/free-flow)
+            rsd[index(c_offset_U, jmax)] = rho_u(x, jmax) - rho_u(x, jmax - 1);
         }
+        diag[index(c_offset_U, jmax)] = 0; // Algebraic constraint
     }
 
-    for (size_t j = jmin; j <= jmax; j++) {
-        //----------------------------------------------
-        //         left boundary
-        //----------------------------------------------
-
-        if (j == 0) {
-            // these may be modified by a boundary object
-
-            // Continuity. This propagates information right-to-left, since
-            // rho_u at point 0 is dependent on rho_u at point 1, but not on
-            // mdot from the inlet.
-            rsd[index(c_offset_U,0)] =
-                -(rho_u(x,1) - rho_u(x,0))/m_dz[0]
-                -(density(1)*V(x,1) + density(0)*V(x,0));
-
-            // the inlet (or other) object connected to this one will modify
-            // these equations by subtracting its values for V, T, and mdot. As
-            // a result, these residual equations will force the solution
-            // variables to the values for the boundary object
-            rsd[index(c_offset_V,0)] = V(x,0);
-            rsd[index(c_offset_T,0)] = T(x,0);
-            if (m_usesLambda) {
-                rsd[index(c_offset_L, 0)] = -rho_u(x, 0);
-            } else {
-                rsd[index(c_offset_L, 0)] = lambda(x, 0);
-                diag[index(c_offset_L, 0)] = 0;
+    // j0 and j1 are constrained to only interior points
+    size_t j0 = std::max<size_t>(jmin, 1);
+    size_t j1 = std::min(jmax, m_points - 2);
+    if (m_usesLambda == true) { // "axisymmetric-flow"
+        for (size_t j = j0; j <= j1; j++) { // interior points
+            // For "axisymmetric-flow", the continuity equation  propagates the
+            // mass flow rate information to the left (j+1 -> j) from the value
+            // specified at the right boundary. The lambda information propagates
+            // in the opposite direction.
+            rsd[index(c_offset_U,j)] = -(rho_u(x,j+1) - rho_u(x,j))/m_dz[j]
+                                    -(density(j+1)*V(x,j+1) + density(j)*V(x,j));
```

Comment:
Suggested indentation:
```suggestion
            rsd[index(c_offset_U,j)] = -(rho_u(x,j+1) - rho_u(x,j))/m_dz[j]
                                       - (density(j+1)*V(x,j+1) + density(j)*V(x,j));
```